### ISSUE TITLE
Upgrade to pgAdmin4 v7 leads to fatal error - duplicate column name: shared

### ISSUE THREAD

**Issue by omatai** [pgadmin-org/pgadmin4#6195]
**_Please note that security bugs or issues should be reported to security@pgadmin.org._**

**Describe the bug**

See [this description](https://stackoverflow.com/questions/76115370/how-to-recover-from-pgadmin-4-upgrade). On Windows 10, after uninstall of pgAdmin 4, and reinstall of pgAdmin 4 v7, pgAdmin 4 consistently fails to connect to server on startup. Full uninstall/reinstall of entire PostgreSQL bundle makes no difference. Offending error seems to be:

> 2023-04-26 19:28:32,904: ERROR  pgadmin:    Database migration failed
> 2023-04-26 19:28:33,031: ERROR  pgadmin:    Traceback (most recent call last):
>   File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\base.py", line 1964, in _exec_single_context
> self.dialect.do_execute(
>   File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\default.py", line 748, in do_execute
> cursor.execute(statement, parameters)
> sqlite3.OperationalError: duplicate column name: shared

**To Reproduce**

1. Go to Control Panel -> Programs -> Programs and Features
2. Select pgAdmin 4 (or the pgAdmin 4 component of the PostgreSQL install) and uninstall it
3. Download and install the latest pgAdmin 4 v7 using x86-64 .exe installer
4. Start pgAdmin 4 v7 -> it will fail to connect to the server

**Expected behavior**

A server connection should be established and pgAdmin 4 should open up in a browser window

**Error message**

2023-04-27 12:26:29,740: ERROR	pgadmin:	Database migration failed
2023-04-27 12:26:29,740: ERROR	pgadmin:	Traceback (most recent call last):
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\base.py", line 1964, in _exec_single_context
    self.dialect.do_execute(
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\default.py", line 748, in do_execute
    cursor.execute(statement, parameters)
sqlite3.OperationalError: duplicate column name: shared

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\web\pgadmin\__init__.py", line 388, in upgrade_db
    db_upgrade(app)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\web\pgadmin\setup\db_upgrade.py", line 25, in db_upgrade
    flask_migrate.upgrade(migration_folder)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\flask_migrate\__init__.py", line 111, in wrapped
    f(*args, **kwargs)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\flask_migrate\__init__.py", line 200, in upgrade
    command.upgrade(config, revision, sql=sql, tag=tag)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\command.py", line 378, in upgrade
    script.run_env()
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\script\base.py", line 576, in run_env
    util.load_python_file(self.dir, "env.py")
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\util\pyfiles.py", line 94, in load_python_file
    module = load_module_py(module_id, path)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\util\pyfiles.py", line 110, in load_module_py
    spec.loader.exec_module(module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\web\pgadmin\setup\..\..\migrations\env.py", line 95, in <module>
    run_migrations_online()
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\web\pgadmin\setup\..\..\migrations\env.py", line 87, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\runtime\environment.py", line 868, in run_migrations
    self.get_context().run_migrations(**kw)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\runtime\migration.py", line 622, in run_migrations
    step.migration_fn(**kw)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\web\migrations\versions\a091c9611d20_.py", line 27, in upgrade
    op.add_column('server', sa.Column('shared', sa.Boolean()))
  File "<string>", line 8, in add_column
  File "<string>", line 3, in add_column
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\operations\ops.py", line 2044, in add_column
    return operations.invoke(op)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\operations\base.py", line 401, in invoke
    return fn(self, operation)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\operations\toimpl.py", line 154, in add_column
    operations.impl.add_column(table_name, column, schema=schema, **kw)
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\ddl\impl.py", line 322, in add_column
    self._exec(base.AddColumn(table_name, column, schema=schema))
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\alembic\ddl\impl.py", line 193, in _exec
    return conn.execute(  # type: ignore[call-overload]
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\base.py", line 1414, in execute
    return meth(
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\sql\ddl.py", line 181, in _execute_on_connection
    return connection._execute_ddl(
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\base.py", line 1526, in _execute_ddl
    ret = self._execute_context(
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\base.py", line 1842, in _execute_context
    return self._exec_single_context(
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\base.py", line 1983, in _exec_single_context
    self._handle_dbapi_exception(
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\base.py", line 2326, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\base.py", line 1964, in _exec_single_context
    self.dialect.do_execute(
  File "C:\Users\Paul\AppData\Local\Programs\pgAdmin 4\v7\python\lib\site-packages\sqlalchemy\engine\default.py", line 748, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) duplicate column name: shared
[SQL: ALTER TABLE server ADD COLUMN shared BOOLEAN]
(Background on this error at: https://sqlalche.me/e/20/e3q8)

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro
 - Version: 22H2
 - Mode: Desktop
 - Browser (did not invoke browser)
 - Package type: ??

**Comment by akshay-joshi** [comment]
Hi @omatai 

It seems to uninstall/reinstall the PostgreSQL version and the community version pgAdmin corrupted the configuration database file. It would be good to take a backup of that file `%APPDATA%/pgadmin/pgadmin4.db` and remove it. After that please relaunch pgAdmin 4.

**Comment by disantimaxi** [comment]
Hi all, I've tried uninstalling PgAdmin4 v7, then install PgAdmin4 v6.21 and I got another different message but I could pass through. Then I've uninstalled the v6.21 and installed v7 back again.

This was the other message when I've installed v6.21:

![image](https://user-images.githubusercontent.com/19420841/235276876-976e24b5-57c4-4222-873b-c5c6128f2719.png)

**Comment by BinarySpike** [comment]
I installed v7 then uninstalled v4 and started getting this error.  Removing the file from `%appdata%` was unintuitive as I installed pgadmin for all users.

**Comment by akshay-joshi** [comment]
No response from the author, hence closing it.

**Comment by omatai** [comment]
Can I respectfully suggest that an unresponsive author is not a good reason to close a bug? My wife started urgent cancer treatment 48 hours after I filed this bug. Life has turned on its head. I have not had cause to go near pgAdmin4 since I submitted it. Either it's a bug, or it's not, independent of authorship. I may or may not ever find time to read the advice here and see if it fixes the bug. That shouldn't stop anyone else from working on it and drawing independent conclusions.
